modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ElmClientCodegen.java

```diff
@@ -527,19 +529,26 @@ private String toOptionalValue(String value) {
 
     private String paramToString(final CodegenParameter param) {
         final String paramName = param.paramName;
-        if (param.isString || param.isUuid) {
+
+        if (param.isString || param.isUuid) { // string
             return paramName;
-        }
-        if (ElmVersion.ELM_018.equals(elmVersion)) {
+        } else if (ElmVersion.ELM_018.equals(elmVersion)) { // Elm 0.18
```

Comment:
All string-like types (for now also binary, file) should be in front of the Elm 0.18 logic.